src/components/Preferences/LocalIdentities.js

```diff
@@ -19,132 +20,250 @@ import {
-  const today = format(Date.now(), 'yyyy-MM-dd')
-
-  return (
-    <React.Fragment>
-      <Headers>
-        <div>Custom label</div>
-        <div>Address</div>
-      </Headers>
-      <List>
-        {identities.map(({ address, name }) => (
-          <Item key={address}>
-            <Label>{name}</Label>
-            <div>
-              <IdentityBadge
-                entity={address}
-                popoverAction={{
-                  label: 'Edit custom label',
-                  onClick: updateLabel(address),
-                }}
-                popoverTitle={<LocalIdentityPopoverTitle label={name} />}
+
+    return (
+      <React.Fragment>
+        <Headers>
+          <div>
+            {!iOS && (
+              <StyledCheckbox
+                checked={allSelected}
+                onChange={onToggleAll}
+                indeterminate={!allSelected && someSelected}
               />
-            </div>
-          </Item>
-        ))}
-      </List>
-      <Controls>
-        <Import onImport={onImport} />
-        {!iOS && (
-          <StyledExport
-            label="Export labels"
-            mode="secondary"
-            download={`aragon-labels_${dao}_${today}.json`}
-            href={downloadHref}
-          >
-            Export
-          </StyledExport>
-        )}
-        <Button label="Remove labels" mode="outline" onClick={onClearAll}>
-          <IconCross /> Remove all labels
-        </Button>
-      </Controls>
-      <Warning />
-    </React.Fragment>
-  )
-}
+            )}
+            Custom label
+          </div>
+          <div>Address</div>
+        </Headers>
+        <List>
+          {identities.map(({ address, name }) => (
+            <Item key={address}>
+              <Label>
+                {!iOS && (
+                  <StyledCheckbox
+                    checked={addressesSelected.get(address)}
+                    onChange={onToggleAddress(address)}
+                  />
+                )}
+                {name}
+              </Label>
+              <div>
+                <IdentityBadge
+                  entity={address}
+                  popoverAction={{
+                    label: 'Edit custom label',
+                    onClick: updateLabel(address),
+                  }}
+                  popoverTitle={<LocalIdentityPopoverTitle label={name} />}
+                />
+              </div>
+            </Item>
+          ))}
+        </List>
+        <Controls>
+          <Import onImport={onImport} />
+          {!iOS && (
+            <StyledExport
+              label="Export labels"
+              mode="secondary"
+              download={`aragon-labels_${dao}_${today}.json`}
+              href={someSelected ? downloadHref : undefined}
+              disabled={!someSelected}
+            >
+              Export
+            </StyledExport>
+          )}
+          <Button label="Remove labels" mode="outline" onClick={onClearAll}>
```

Comment:
No need for `label` (or `title` − `label` is only supported by `ButtonIcon`) since we already have text in the button:

```suggestion
          <Button mode="outline" onClick={onClearAll}>
```